cstr/src/lib.rs

```diff
@@ -176,14 +184,22 @@ pub fn to_char_ptr(s: &str) -> (*mut Char, Vec<u8>) {
 /// # Safety
 /// This writes to a raw c-pointer, which requires unsafe operations
 #[allow(clippy::cast_possible_truncation, clippy::cast_possible_wrap)]
-pub unsafe fn write_string_to_buffer(message: &str, len: usize, output_ptr: *mut WideChar) -> i32 {
-    let len = std::cmp::min(message.len(), len - 1);
-    let mut v = to_widechar_vec(&message[..len]);
-    v.push(0);
-    unsafe {
-        copy_nonoverlapping(v.as_ptr(), output_ptr, len);
+pub unsafe fn write_string_to_buffer(
+    message: &str,
+    len: isize,
```

Comment:
`isize` is the maximum we can write according to the spec if I'm reading it correctly.  I'll use `try_from`, but it can never panic in the else clause because we check if the length is 0. `len` can never be negative because of the `(len - 1) as usize`, which will convert to `usize::MAX` if it is negative.